tutorial/mpi4py/mpi4py_portable.py

```diff
@@ -0,0 +1,123 @@
+
+from eessi.testsuite import hooks
+from eessi.testsuite.constants import SCALES, COMPUTE_UNIT, CPU
+from eessi.testsuite.utils import find_modules
+
+
+# This python decorator indicates to ReFrame that this class defines a test
+# Our class inherits from rfm.RunOnlyRegressionTest, since this test does not have a compilation stage
+# https://reframe-hpc.readthedocs.io/en/stable/regression_test_api.html#reframe.core.pipeline.RunOnlyRegressionTest
+@rfm.simple_test
+class EESSI_MPI4PY(rfm.RunOnlyRegressionTest):
+    # Programming environments are only relevant for tests that compile something
+    # Since we are testing existing modules, we typically don't compile anything and simply define
+    # 'default' as the valid programming environment
+    # https://reframe-hpc.readthedocs.io/en/stable/regression_test_api.html#reframe.core.pipeline.RegressionTest.valid_prog_environs
+    valid_prog_environs = ['default']
+
+    # Typically, we list here the name of our cluster as it is specified in our ReFrame configuration file
+    # https://reframe-hpc.readthedocs.io/en/stable/regression_test_api.html#reframe.core.pipeline.RegressionTest.valid_systems
+    valid_systems = ['snellius']
+
+    # ReFrame will generate a test for each module
+    # https://reframe-hpc.readthedocs.io/en/stable/regression_test_api.html#reframe.core.builtins.parameter
+    module_name = parameter(find_modules('mpi4py'))
+
+    # ReFrame will generate a test for each scale
+    scale = parameter(SCALES.keys())
+
+    # Our script has two arguments, --n_iter and --n_warmup. By defining these as ReFrame variables, we can
+    # enable the end-user to overwrite their value on the command line when invoking ReFrame.
+    # Note that we don't typically expose ALL variables, especially if a script has many - we expose
+    # only those that we think an end-user might want to overwrite
+    # Number of iterations to run (more iterations takes longer, but results in more accurate timing)
+    # https://reframe-hpc.readthedocs.io/en/stable/regression_test_api.html#reframe.core.builtins.variable
+    n_iterations = variable(int, value=1000)
+
+    # Similar for the number of warmup iterations
+    n_warmup = variable(int, value=100)
+
+    # Define which executable to run
+    # https://reframe-hpc.readthedocs.io/en/stable/regression_test_api.html#reframe.core.pipeline.RegressionTest.executable
+    executable = 'python3'
+
+    # Define which options to pass to the executable
+    # https://reframe-hpc.readthedocs.io/en/stable/regression_test_api.html#reframe.core.pipeline.RegressionTest.executable_opts
+    executable_opts = ['mpi4py_reduce.py', '--n_iter', f'{n_iterations}', '--n_warmup', f'{n_warmup}']
+
+    # Temporarily define postrun_cmds to make it easy to find out memory useage
+    postrun_cmds = [
+        'MAX_MEM_IN_BYTES=$(cat /sys/fs/cgroup/memory/$(</proc/self/cpuset)/memory.max_usage_in_bytes)',
```

Comment:
Also, I guess I could do the `&& cat ...` by appending that as additional `executable_opts`. So that indeed gives people a way to do it also with `srun`

Reply:
the problem is that `srun` can only run a single command, hence the `srun bash -c 'blabla && boemboem'` trick.
not sure how to fit that into an `executable_opts`?

Reply:
oh good point, it needs the `bash` as executable. Ugh... that's messy.

Ok different idea: we just state it only works with mpirun. I'll make a note that if they _have_ to run with `srun` they can do `--dry-run` and manually modify the job script to use your 'trick' :)